tracing-journald/src/lib.rs

```diff
@@ -109,6 +114,48 @@ impl Subscriber {
         self.field_prefix = x;
         self
     }
+
+    #[cfg(not(unix))]
+    fn send_payload(&self, _opayload: &[u8]) -> std::io::Result<()> {
+        Err(io::Error::new(
+            io::ErrorKind::Unsupported,
+            "journald not supported on non-Unix",
+        ))
+    }
+
+    #[cfg(unix)]
+    fn send_payload(&self, payload: &[u8]) -> std::io::Result<usize> {
+        self.socket.send(payload).or_else(|error| {
+            if Some(libc::EMSGSIZE) == error.raw_os_error() {
+                self.send_large_payload(payload)
+            } else {
+                Err(error)
+            }
+        })
+    }
+
+    #[cfg(not(target_os = "linux"))]
+    fn send_large_payload(&self, _payload: &[u8]) -> std::io::Result<usize> {
+        Err(io::Error::new(
+            io::ErrorKind::Unsupported,
+            "Large payloads not supported on non-Linux OS",
+        ))
+    }
```

Comment:
honestly, i'm a little unsure why we go out of our way to support `#[cfg(unix)]` environments that are not linux. is it even _possible_ to use `journald` on BSD?

not a blocker for this PR, because we already do this, just a thought.

Reply:
I didn't actually think much about this… I used `cfg` to tag where things build, not necessarily where they run :slightly_smiling_face:  

I don't know whether journald works on BSD; I don't know much about other Unix systems, and I must admit that I don't care enough to find out :innocent: 

I can also use a simpler `cfg` matrix which mostly stubs out everything on non-Linux, but I think I'd prefer @Ralith's feedback about this as well :slightly_smiling_face:

Reply:
I don't have terribly strong opinions about how we behave on non-Linux unix so long as it's graceful and complexity is limited; I think it's mostly academic.